### PR TITLE
Substitute category (deprecate) with tag

### PR DESCRIPTION
**Pull request by sourishkrout**
Users have reported that `tags` are self-explanatory, whereas `categories` are not.

1. In the stable CLI interface, let's move off `category` without interrupting folks.
2. In `beta`, there will only be a `tag`.

### CODE REVIEW

**Comment by sourishkrout** [comment]
Another review, please, @pastuxso.

File: internal/config/config_filter.go [hunk 1]
```diff
@@ -34,7 +34,7 @@ type FilterDocumentEnv struct {
 // Without it, all variables start with capitalized letters.
 type FilterBlockEnv struct {
 	Background             bool     `expr:"background"`
-	Categories             []string `expr:"categories"`
+	Tags                   []string `expr:"tags"`
```
Comment (adambabik) [inline internal/config/config_filter.go:37]: Suggestion: fields are sorted alphabetically in this struct.

File: internal/config/autoconfig/autoconfig.go [hunk 2]
```diff
@@ -171,7 +171,7 @@ func getProjectFilters(c *config.Config) ([]project.Filter, error) {
 			filters = append(filters, project.Filter(func(t project.Task) (bool, error) {
 				env := config.FilterBlockEnv{
 					Background: t.CodeBlock.Background(),
-					Categories: t.CodeBlock.Categories(),
+					Tags:       t.CodeBlock.Tags(),
```
Comment (adambabik) [inline internal/config/autoconfig/autoconfig.go:174]: Same suggestion about sorting fields.

File: pkg/document/block.go [hunk 3]
```diff
@@ -102,12 +102,16 @@ func (b *CodeBlock) Background() bool {
 	return val
 }
 
-func (b *CodeBlock) Categories() []string {
-	cat, ok := b.Attributes()["category"]
+func (b *CodeBlock) Tags() []string {
+	tag, ok := b.Attributes()["category"]
+	if !ok {
+		tag, ok = b.Attributes()["tag"]
```
Comment (adambabik) [inline pkg/document/block.go:108]: We don't care about a case where both are set?

File: pkg/document/frontmatter.go [hunk 4]
```diff
@@ -58,6 +58,7 @@ type Frontmatter struct {
 	Shell        string         `yaml:"shell"`
 	Cwd          string         `yaml:"cwd"`
 	Category     string         `yaml:"category"`
```
Comment (adambabik) [inline pkg/document/frontmatter.go:60]: Should `Category` be marked as deprecated?

File: pkg/api/proto/runme/parser/v1/parser.proto [hunk 5]
```diff
@@ -95,6 +95,7 @@ message Frontmatter {
   FrontmatterRunme runme = 4;
   string category = 5;
```
Comment (adambabik) [inline pkg/api/proto/runme/parser/v1/parser.proto:96]: Should we comment `category` as deprecated?

File: internal/cmd/run.go [hunk 6]
```diff
@@ -87,35 +92,35 @@ func runCmd() *cobra.Command {
...
+							fmTags := resolveFrontmatterTags(fm)
 							match := false
-							if len(fmCategories) > 0 && containsCategories(fmCategories, cmdCategories) {
-								if len(blockCategories) == 0 {
+							if len(fmTags) > 0 && containsCategories(fmTags, cmdTags) {
+								if len(blockTags) == 0 {
 									match = true
 								} else {
-									match = containsCategories(fmCategories, blockCategories)
+									match = containsCategories(fmTags, blockTags)
```
Comment (pastuxso) [inline internal/cmd/run.go:116]: What do you think on renaming the function to `containsTags`?

**Review by pastuxso — CHANGES_REQUESTED** [review]
Works fine, I left you a minor comment.

**Comment by sourishkrout** [comment]
Whoops, I somehow overlooked @adambabik's comments. I am addressing them now.

File: pkg/document/block.go (hunk 3 above)
Comment (sourishkrout) [reply pkg/document/block.go:108]: Right, I am combining them instead.

**Pull request merged**